lib/routes/user/add_funds/add_funds_page.dart

```diff
@@ -43,71 +51,89 @@ class AddFundsState extends State<AddFundsPage> {
-                      appBar: new AppBar(
-                        iconTheme: theme.appBarIconTheme,
-                        textTheme: theme.appBarTextTheme,
-                        backgroundColor: theme.BreezColors.blue[500],
-                        leading: backBtn.BackButton(),
-                        title: new Text(
-                          _title,
-                          style: theme.appBarTextStyle,
-                        ),
-                        elevation: 0.0,
-                      ),
-                      body: new Container(
-                        child: Material(
-                            child: getBody(
-                                context,
-                                accSnapshot.data,
-                                snapshot.data,
-                                snapshot.hasError
-                                    ? "Failed to retrieve an address from Breez server\nPlease check your internet connection."
-                                    : null)),
-                      )),
-                );
-              });
-        });
+              builder: (BuildContext context, AsyncSnapshot<AddFundResponse> response) {
+                if (!response.hasData && !response.hasError) {
+                  return Center(child: Loader(color: theme.BreezColors.white[400]));
+                }
+                return getBody(context, account.data, response.data,
+                    response.hasError ? "Failed to retrieve an address from Breez server\nPlease check your internet connection." : null);
+              },
+            );
+          },
+        ),
+      ),
+    );
   }
 
-  Widget getBody(BuildContext context, AccountModel account,
-      AddFundResponse response, String error) {
+  Widget getBody(BuildContext context, AccountModel account, AddFundResponse response, String error) {
     var unconfirmedTxID = account?.swapFundsStatus?.unconfirmedTxID;
     bool waitingDepositConfirmation = unconfirmedTxID?.isNotEmpty == true;
 
     String errorMessage;
     if (error != null) {
       errorMessage = error;
     } else if (account == null || account.bootstraping) {
-      errorMessage =
-          'You\'d be able to add funds after Breez is finished bootstrapping.';
-    } else if (unconfirmedTxID?.isNotEmpty == true ||
-        account.processingWithdrawal) {
+      errorMessage = 'You\'d be able to add funds after Breez is finished bootstrapping.';
+    } else if (unconfirmedTxID?.isNotEmpty == true || account.processingWithdrawal) {
       errorMessage =
           'Breez is processing your previous ${waitingDepositConfirmation || account.processingBreezConnection ? "deposit" : "withdrawal"}. You will be able to add more funds once this operation is completed.';
     } else if (response != null && response.errorMessage.isNotEmpty) {
       errorMessage = response.errorMessage;
+    } else if (_moonPayOrder?.address != null &&
+        account?.addedFundsReply?.unConfirmedAddresses
```

Comment:
Although the `addedFundsReply` is public field it wasn't the intention as it is a protobuf structure.
The access should be via the `SwapFundStatus` that wraps it.
I would add a getter there for `unconfirmedAddresses` same as the `refundableAddresses` that exists there.

Reply:
https://github.com/breez/breezmobile/pull/162/commits/eb57e859db19800afe4054b9d63ef1bcb0617671